From 8688f06f5653f3509dc35d52082bbee887522eab Mon Sep 17 00:00:00 2001
From: Bill Schwartz <whs@us.ibm.com>
Date: Mon, 7 Dec 2015 09:10:50 -0600
Subject: [PATCH] No RCD CNTRL words for DDR4 RDIMMs

Return 0 for DDR4 RDIMMS for ATTR_SPD_DIMM_RCD_CNTL_WORD_0_15

Change-Id: I15d99e5b87035f30a9a068cdeb52d7db4be37651
RTC: 137707
---
 src/usr/hwpf/plat/fapiPlatAttributeService.C | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/usr/hwpf/plat/fapiPlatAttributeService.C b/src/usr/hwpf/plat/fapiPlatAttributeService.C
index 82e7afd..25953f2 100644
--- a/src/usr/hwpf/plat/fapiPlatAttributeService.C
+++ b/src/usr/hwpf/plat/fapiPlatAttributeService.C
@@ -1800,6 +1800,26 @@ fapi::ReturnCode fapiPlatGetRCDCntlWord015(const fapi::Target * i_pFapiTarget,
 
         if (l_dimmType == ENUM_ATTR_SPD_MODULE_TYPE_RDIMM)
         {
+            ATTR_SPD_DRAM_DEVICE_TYPE_Type l_spd_dramtype =
+                                         ENUM_ATTR_SPD_DRAM_DEVICE_TYPE_DDR3;
+
+            l_rc = FAPI_ATTR_GET(ATTR_SPD_DRAM_DEVICE_TYPE,
+                                 i_pFapiTarget,
+                                 l_spd_dramtype);
+            if (l_rc)
+            {
+                FAPI_ERR("fapiPlatGetRCDCntlWord015: Error from get"
+                         " ATTR_SPD_DRAM_DEVICE_TYPE");
+                break;
+            }
+
+            // Not defined for DDR4
+            if (ENUM_ATTR_SPD_DRAM_DEVICE_TYPE_DDR4 == l_spd_dramtype)
+            {
+                o_val = 0;
+                break;
+            }
+
             TARGETING::Target* l_pTarget = NULL;
             l_rc = getTargetingTarget(i_pFapiTarget, l_pTarget, TARGETING::TYPE_DIMM);
             if (l_rc)
-- 
1.8.2.2

